cmd/installer/cli/restore.go

```diff
@@ -81,446 +81,389 @@ const (
-					logrus.Infof("Before you can restore, you must remove the existing installation.")
-					logrus.Infof("You can do this by running the following command:")
-					logrus.Infof("\n  sudo ./%s reset\n", name)
-					os.Exit(1)
-				}
+	if flags.airgapBundle != "" {
+		logrus.Debugf("checking airgap bundle matches binary")
+		if err := checkAirgapMatches(flags.airgapBundle); err != nil {
+			return err // we want the user to see the error message without a prefix
+		}
+	}
 
-				if !s3BackupStoreHasData(s3Store) {
-					logrus.Infof("You'll be guided through the process of restoring %s from a backup.\n", name)
-					logrus.Info("Enter information to configure access to your backup storage location.\n")
+	// if the user wants to resume, check if a backup has already been picked.
+	var backupToRestore *disasterrecovery.ReplicatedBackup
+	if state != ecRestoreStateNew {
+		logrus.Debugf("getting backup from restore state")
+		var err error
+		backupToRestore, err = getBackupFromRestoreState(ctx, flags.airgapBundle != "")
+		if err != nil {
+			return fmt.Errorf("unable to resume: %w", err)
+		}
+		if backupToRestore != nil {
+			completionTimestamp := backupToRestore.GetCompletionTimestamp().Format("2006-01-02 15:04:05 UTC")
+			logrus.Infof("Resuming restore from backup %q (%s)\n", backupToRestore.GetName(), completionTimestamp)
 
-					promptForS3BackupStore(s3Store)
-				}
-				s3Store.prefix = strings.TrimPrefix(s3Store.prefix, "/")
+			if err := overrideRuntimeConfigFromBackup(flags.localArtifactMirrorPort, *backupToRestore); err != nil {
+				return fmt.Errorf("unable to override runtime config from backup: %w", err)
+			}
+		}
+	}
 
-				skipStoreValidationFlag, err := cmd.Flags().GetBool("skip-store-validation")
-				if err != nil {
-					return fmt.Errorf("unable to get skip-store-validation flag: %w", err)
-				}
+	// If the installation is available, we can further augment the runtime config from the installation.
+	rc, err := getRuntimeConfigFromInstallation(cmd)
+	if err != nil {
+		logrus.Debugf(
+			"Unable to get runtime config from installation, this is expected if the installation is not yet available (restore state=%s): %v",
+			state, err,
+		)
+	} else {
+		runtimeconfig.Set(rc)
+	}
 
-				if !skipStoreValidationFlag {
-					logrus.Debugf("validating backup store configuration")
-					if err := validateS3BackupStore(s3Store); err != nil {
-						return fmt.Errorf("unable to validate backup store: %w", err)
-					}
-				}
+	os.Setenv("KUBECONFIG", runtimeconfig.PathToKubeConfig())
+	os.Setenv("TMPDIR", runtimeconfig.EmbeddedClusterTmpSubDir())
 
-				logrus.Debugf("configuring network manager")
-				if err := configureNetworkManager(cmd.Context()); err != nil {
-					return fmt.Errorf("unable to configure network manager: %w", err)
-				}
+	opts := addonsApplierOpts{
+		assumeYes:    flags.assumeYes,
+		license:      "",
+		airgapBundle: flags.airgapBundle,
+		overrides:    "",               // TODO: why not set this?
```

Comment:
should we add this to the restore command? im not sure the restore will apply the overrides in all cases.

Reply:
I think we should in case there are overrides for openebs or velero, no?